controllers/teleport_master_controller/teleport_master_controller.cpp

```diff
@@ -0,0 +1,139 @@
+    const double ySize     = tempNode->getField("ySize")->getSFFloat();
+
+    // Get the time step of the current world.
+    int timeStep = int(supervisor.getBasicTimeStep());
+
+    // Generate random seed
+    std::random_device rd;   // Will be used to obtain a seed for the random number engine
+    std::mt19937 gen(rd());  // Standard mersenne_twister_engine seeded with rd()
+
+    // Generate distributions for random number systems in the following loop
+    std::uniform_real_distribution<> xDistrib(-xSize * 0.5, xSize * 0.5);
+    std::uniform_real_distribution<> yDistrib(-ySize * 0.5, ySize * 0.5);
+    std::uniform_real_distribution<> rotDistrib(0.0, rotations.size() - 1.0);
+
+    while (supervisor.step(timeStep) != -1) {
+
+        // Declare a vector of positions that will be saved as they are randomly generated, to be later
+        // applied to each robot
+        std::vector<std::array<double, 3>> positions;
+        // Loop through every robot
+        for (size_t i = 0; i < otherRobotsNodes.size(); i++) {
+            // Assume there is no collision
+            bool collision = false;
+            // newPos will be a "Proposed location" for a robot to teleport to
+            std::array<double, 3> newPos;
+            do {
+                collision = false;
+                // Generate a new random location
+                newPos[0] = xDistrib(gen);
+                newPos[1] = yDistrib(gen);
+                newPos[2] = zHeight;
+
+                // Loop through the vector of existing proposed locations and see if the new one is going to
+                // collide with any of them
+                for (const auto& testPos : positions) {
+                    const double distance =
+                        std::sqrt(std::pow((newPos[1] - testPos[1]), 2) + std::pow((newPos[0] - testPos[0]), 2));
+                    if (distance < minDistance) {
+                        collision = true;
+                    }
+                }
+                // Loop until a proposed location has been found that doesn't clash with the existing ones
+            } while (collision);
+            // Finally add the proposed location in as a confirmed position
+            positions.emplace_back(newPos);
+        }
+
+        // Loop through every robot
+        for (size_t i = 0; i < otherRobotsNodes.size(); i++) {
+            // Grab translation field of the robot to modify
+            webots::Field* target_translation_field = otherRobotsNodes[i]->getField("translation");
+            // Grab the current rotation field of the robot to modify
+            webots::Field* target_rotation_field = otherRobotsNodes[i]->getField("rotation");
+
+            // There will be a position for every robot in the positions vector
+            target_translation_field->setSFVec3f(positions[i].data());
+            // Apply new rotation
+            target_rotation_field->setSFRotation(rotations[rotDistrib(gen)].data());
```

Comment:
I just noticed that you only use each of these variables once. I would just do this
```suggestion
            // There will be a position for every robot in the positions vector
            otherRobotsNodes[i]->getField("translation")->setSFVec3f(positions[i].data());
            // Apply new rotation
            otherRobotsNodes[i]->getField("rotation")->setSFRotation(rotations[rotDistrib(gen)].data());
```

Reply:
I was considering that, I decided to leave it split up for readability but I can merge them if that's more suitable